invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -168,7 +169,7 @@ class SubjectSchema43(Schema):
     subjectScheme = fields.Str(attribute="scheme")
 
 
-class DataCite43Schema(Schema):
+class DataCite43Schema(BaseSchema):
```

Comment:
In order to have a `post_dump` pipeline, other schemas must inherit from `BaseSchema` as well.

Question: shouldn't `BaseSchema` be higher in the hierarchy? E.g. `flask-resources`.

ping @ppanero  @slint

Reply:
I'm a bit against the current use/naming of `BaseSchema`. It currently contains "common metadata helper methods" that are used in sub-classing serializer schemas for tansforming/formatting fields.

Now, we want to introduce our "plugin" mixin which adds `post_dump`/`pre_dump` methods, which is more of a configuration/functional/plugin "utility" rather than centralizing common metadata rules. See my comment on the class below on how we could maybe proceed.

---

(separate note for the "common metadata helper methods")

I'm certain also that as we build more format serializers the common metadata rules will diverge a lot and become a bit messy... So in this case I would wait until we first build them, and afterwards deduplicate/centralize logic.

Reply:
> I'm a bit against the current use/naming of `BaseSchema`. It currently contains "common metadata helper methods" that are used in sub-classing serializer schemas for tansforming/formatting fields.
> ## Now, we want to introduce our "plugin" mixin which adds `post_dump`/`pre_dump` methods, which is more of a configuration/functional/plugin "utility" rather than centralizing common metadata rules.
> 
> (separate note for the "common metadata helper methods")
> 
> I'm certain also that as we build more format serializers the common metadata rules will diverge a lot and become a bit messy... So in this case I would wait until we first build them, and afterwards deduplicate/centralize logic.

In that case, are you suggesting to duplicate the logic in all the schemas we currently have and then deduplicate it when we figure out a better structure for schemas/serializers?

Reply:
While I do agree on the `BaseClass` naming, we did this to deduplicate code from 4 serializers. If in the future the deduplicate we can re-think it. There can be other ways (mixins, etc.) I'll leave the choice of method to you. But I'd not go to duplicate code...

Reply:
> In order to have a `post_dump` pipeline, other schemas must inherit from `BaseSchema` as well.

Why is that? The [BaseSchema](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/serializers/schemas.py#L12) class only contains a few helper methods at the moment. Joining what @slint said, it now feels like more a helper, utility, a plugin, than a base class (which maybe made sense before!).

If I interpret well what Alex suggests, that should be converted to a Mixin or injected via contructor (if possible) to compose functionalities, and the convert the children to:
```
class ASchema(Schema, CommonFieldsMixin)
```

> Question: shouldn't `BaseSchema` be higher in the hierarchy? E.g. `flask-resources`.
Maybe I am not looking at the right BaseSchema class, that's why I have doubts with my question above, but it contains a few fields of the RDM data model, so it should stay here.

Reply:
+1 to CommonFieldsMixin